redis_cache/redis_cache.py

```diff
@@ -68,3 +68,25 @@ def _default_signature_generator(*args, **kwargs):
             lambda x: x != '', [parsed_args, parsed_kwargs]
         )
         return ','.join(parsed)
+
+
+def _argument_to_string(arg):
+    if arg.__class__.__module__ == '__builtin__':
+        return str(arg)
+
+    # for backwards-compatibility: if the object defines a custom __str__ method
+    # use it instead of the default approach based on the class-name
+    string_representation = str(arg)
+    if string_representation != object.__str__(arg):
+        return string_representation
+
+    try:
+        instance_namespace = arg.__class__.__name__
+        instance_state = sorted((field, _argument_to_string(value))
```

Comment:
So this generates a cache key based on the object's name and the instance variables that it contains? And it gets those instance variables recursively, so if the instance variables are objects with state themselves, they also get expressed?